src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -18,61 +19,52 @@ def __init__(
         tc: TablesCrawler,
         ws: WorkspaceClient,
         backend: SqlBackend,
-        default_catalog=None,
-        database_to_catalog_mapping: dict[str, str] | None = None,
+        tm: TableMapping,
     ):
         self._tc = tc
         self._backend = backend
         self._ws = ws
-        self._database_to_catalog_mapping = database_to_catalog_mapping
-        self._default_catalog = self._init_default_catalog(default_catalog)
+        self._tm = tm
         self._seen_tables: dict[str, str] = {}
 
-    @staticmethod
-    def _init_default_catalog(default_catalog):
-        if default_catalog:
-            return default_catalog
-        else:
-            return "ucx_default"  # TODO : Fetch current workspace name and append it to the default catalog.
-
     def migrate_tables(self):
         self._init_seen_tables()
+        mapping_rules = self._get_mapping_rules_dict()
         tasks = []
         for table in self._tc.snapshot():
-            target_catalog = self._default_catalog
-            if self._database_to_catalog_mapping:
-                target_catalog = self._database_to_catalog_mapping[table.database]
-            tasks.append(partial(self._migrate_table, target_catalog, table))
-        _, errors = Threads.gather("migrate tables", tasks)
-        if len(errors) > 0:
-            # TODO: https://github.com/databrickslabs/ucx/issues/406
-            # TODO: pick first X issues in the summary
-            msg = f"Detected {len(errors)} errors: {'. '.join(str(e) for e in errors)}"
-            raise ValueError(msg)
-
-    def _migrate_table(self, target_catalog: str, table: Table):
-        sql = table.uc_create_sql(target_catalog)
-        logger.debug(f"Migrating table {table.key} to using SQL query: {sql}")
-        target = f"{target_catalog}.{table.database}.{table.name}".lower()
-
-        if self._table_already_upgraded(target):
-            logger.info(f"Table {table.key} already upgraded to {self._seen_tables[target]}")
-        elif table.object_type == "MANAGED":
+            rule = mapping_rules.get(table.key)
+            if not rule:
+                logger.info(f"Skipping table {table.key} table doesn't exist in the mapping table.")
+                continue
+            tasks.append(partial(self._migrate_table, table, rule.as_uc_table_key))
```

Comment:
```suggestion
            if rule.skip_table: 
              continue
            tasks.append(partial(self._migrate_table, table, rule.as_uc_table_key))
```
also check if the table is skipped by `databricks labs ucx skip` CLI

alternatively, do that within `_migrate_table`.

Reply:
That's slated for the next PR...